core-blocks/image/edit.js

```diff
@@ -179,31 +198,53 @@ class ImageEdit extends Component {
 		return get( this.props.image, [ 'media_details', 'sizes' ], {} );
 	}
 
+	switchToEditing() {
+		const { url } = this.props.attributes;
+
+		this.props.setAttributes( {
+			url: undefined,
+			alt: undefined,
+			id: undefined,
+			caption: undefined,
+			src: url,
+		} );
+	}
+
 	render() {
 		const { attributes, setAttributes, isLargeViewport, isSelected, className, maxWidth, noticeOperations, noticeUI, toggleSelection } = this.props;
 		const { url, alt, caption, align, id, href, width, height } = attributes;
+		const isExternal = url ? parse( wpApiSettings.root ).host !== parse( url ).host : true;
 
 		const controls = (
 			<BlockControls>
 				<BlockAlignmentToolbar
 					value={ align }
 					onChange={ this.updateAlignment }
 				/>
-
 				<Toolbar>
-					<MediaUpload
-						onSelect={ this.onSelectImage }
-						type="image"
-						value={ id }
-						render={ ( { open } ) => (
-							<IconButton
-								className="components-toolbar__control"
-								label={ __( 'Edit image' ) }
-								icon="edit"
-								onClick={ open }
-							/>
-						) }
-					/>
+					{ true === editorCanUpload() && ! isExternal ? (
```

Comment:
Minor: It's explicit, but I'd prefer it to just be the simpler `editorCanUpload()` truthy check. This is common enough that it stuck out to me that we might be considering some other truthy return values from which we want to distinguish `true`.